### PR TITLE
Implement tag add feature

### LINKED ISSUE #45: As an user, I can create a tag without assigning it to any contact
so that I can find or sort the contacts based on tags later on.

### PR DESCRIPTION
**Pull request by khinkhinn**
Files added:
- TagAddCommandParser.java
- TagAddCommand.java
- TagAddCommandParserTest.java
- TagAddCommandTest.java

Changes:
- ParserUtil.java: change method parseTag to parseTagName, returns a TagName instead of a Tag to be consistent with all other methods in ParserUtil.
- ParserUtilTest.java: change unit tests to use parseTagName instead of parseTag
- TagBuilder: change method withTag to withTagName
- CommandTestUtil.java: added VALID_TAG_NAME, TAG_NAME_DESC and INVALID_TAG_NAME

### CODE REVIEW

**Comment by khinkhinn** [comment]
sry dont merge yet i need to add tests

**Review by yeo-kexin — COMMENTED** [review]
small error, but other than that LGTM

File: src/main/java/seedu/address/logic/commands/tagcommands/TagAddCommand.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    public static final String COMMAND_WORD = "tagadd";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a tag to the address book. "
+            + "Parameters: "
+            + PREFIX_TAG + "TAG\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_TAG + "cs2101";
+
+    public static final String MESSAGE_SUCCESS = "New tag added: %1$s";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This tag already exists in the address book";
```
Comment (yeo-kexin) [inline src/main/java/seedu/address/logic/commands/tagcommands/TagAddCommand.java]: i think should be MESSAGE_DUPLICATE_TAG?

Comment (khinkhinn) [reply src/main/java/seedu/address/logic/commands/tagcommands/TagAddCommand.java]: yes thanks

**Review by siqiang-ng — APPROVED** [review]
LGTM

**Review by yeo-kexin — COMMENTED** [review]
LGTM!

**Pull request merged**